### DISCUSSION TITLE
Future Proof Shortcode Calling

### DISCUSSION THREAD

**Discussion by Brixy** [11ty/eleventy#2144] [Q&A] [answered]
What I really like about SSGs is that they are future proof because markdown and HTML files can be rendered with almost any SSG.

I have ported a number of projects from other SSGs to Eleventy. Rewriting the template files was never a problem as Eleventy is so flexible.

Yet, I have encountered problems with shortcodes, for example coming from the Go template language: In some projects I use shortcodes a lot—for images, videos, blockquotes … So I had to manually edit all markdown files to switch to Eleventy.

Do you guys have a suggestion how to make shortcodes future proof? Rewriting the shortcodes is not a problem, but the code for calling a shortcode should be independent from a specific template language.

Thank you very much!

**Comment by Brixy** [comment, marked as answer]
Happy New Year, everybody.

Please let me answer to myself:

I do not use shortcodes at all anymore, because they are too specific to a template language. Content files should be interchangeable between different SSGs, template languages etc.

So this is my current approach:

- Try to use markdown (with some HTML) as much as possible.
- For simple markdown enhancements use regex replacements (e.g. images).
- If a component requires JS anyway let JavaScript enhance the component. (E.g. YouTube videos: I use a modified version of [this script](https://gomakethings.com/how-to-lazy-load-youtube-videos-with-vanilla-javascript/).
- For all other use cases use Eleventy’s [WebC](https://github.com/11ty/webc).

**Comment by sombriks** [comment]
hi Brixy, do you use webce for your layouts?
i am having issues trying to get page.title on my layout. something like this should work, rignt?

```
<!-- header content -->
<head>
    <meta webc:keep charset="utf-8">
    <meta webc:keep name="viewport" content="width=device-width,initial-scale=1">
<!--    <script webc:type="js">-->
<!--        `<title>${page.title}</title>`-->
<!--    </script>-->
    <title @raw="page.title"></title>
<!-- more header content -->
```

Update: 

ok this is close to what i need:
```
<!-- header content -->
    <script webc:type="js" webc:keep webc:is="title">
        title || page.url
    </script>
<!-- header content -->
```

**Reply by Brixy** [reply]
Hi @sombriks!

> hi Brixy, do you use webce for your layouts?

No, I still use Nunjucks for templating, but webC for for small components. Converting templates to webC may be a future option.

Glad you found a solution. [This repo](https://github.com/darthmall/11ty.webc.fun/) may be helpful, too.
